Sample_v3/Samples/SwiftUISimpleChat/UserListView.swift

```diff
@@ -31,16 +31,7 @@ struct UserListView: View {
                         /// Workaround for gestures to work with the whole cell.
                         .contentShape(Rectangle())
                         /// On tap gesture direct message channel will be created and you will be directed back to channel list.
-                        .onTapGesture {
-                            let client = userList.controller.client
-                            let directChatUsers = [client.currentUserId, userList.users[index].id]
-                            let channelController = try! client.channelController(
-                                createDirectMessageChannelWith: Set(directChatUsers),
-                                extraData: .init()
-                            )
-                            channelController.synchronize()
-                            presentationMode.wrappedValue.dismiss()
-                        }
```

Comment:
@VojtaStavik `createDirectMessageChannel` does the same thing that was happening in `UserListView` when a user is selected. The thing is `UserListView` can be opened not only from channel-list, but also from channel-member-list. In the former case we don't want to create 1-1 channel but add selected user to the channel instead.

Answering your question - yes, we create a new channel and throw the controller away. It should open the created 1-1 channel but it was not implemented so I just preserved the current behavior.

Reply:
@evsaev If you could fix it, it would be great. The sample code in this form doesn't make much sense and it brings more questions than answers.